helix-term/src/ui/spinner.rs

```diff
@@ -46,22 +55,25 @@ impl Spinner {
     }
 
     pub fn dots(interval: u64) -> Self {
-        Self::new(vec!["⣾", "⣽", "⣻", "⢿", "⡿", "⣟", "⣯", "⣷"], interval)
+        Self::new("⣾⣽⣻⢿⡿⣟⣯⣷", interval)
     }
 
     pub fn start(&mut self) {
         self.start = Some(Instant::now());
     }
 
-    pub fn frame(&self) -> Option<&str> {
-        let idx = (self
-            .start
-            .map(|time| Instant::now().duration_since(time))?
-            .as_millis()
-            / self.interval as u128) as usize
-            % self.count;
-
-        self.frames.get(idx).copied()
+    pub fn frame(&self) -> Option<String> {
+        let idx = if self.count > 1 {
+            (self
+                .start
+                .map(|time| Instant::now().duration_since(time))?
+                .as_millis()
+                / self.interval as u128) as usize
+                % self.count
+        } else {
+            self.start.and(Some(0))?
+        };
+        Some(self.frames.get(idx)?.to_string())
```

Comment:
Is creating a string every time that much cheaper than the old method? At least on windows, allocation is slow due to security features.

Reply:
that's a great point. I was under the impression that `to_string()` was relatively inexpensive, but you're right; it's a lot of allocations. Maybe we should go back to `Vec<&'static str>`.

Reply:
Or instead of vec just one one String and iterate it? Then you could iterate grapheme clusters and return &str

Reply:
An indication helix deals with this elsewhere: https://github.com/helix-editor/helix/blob/master/helix-core/src/graphemes.rs

Reply:
Actually, the original code uses `.copied()` on the string - so I think we're allocating anyway.

Reply:
If it was a Vec of String, this line would be

```
self.frames.get(idx).map(|s| s.as_str())
```